garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -259,3 +259,23 @@ function meta:AngleBetween( vec, nrm )
 
 	return math.atan2( self:Determinant( vec, nrm:GetNormalized() ), self:Dot( vec ) )
 end
+
+--[[---------------------------------------------------------
+Recalculates the orthogonality of an up vector according to the
+other vector given as forward direction direction. The call
+changes the self vector assigned as an up direction
+ * self > The up direction being orthogonalized
+ * vF   > The forward direction for the orthogonalization process
+ * bN   > When set to true, normalizes the three unit vectors
+ * vR   > The function returns the right vector on success
+-----------------------------------------------------------]]
+function meta:Orthogonal(vF, bN)

```

Comment:
I don't think this should be part of the lib - it's arbitrary for the up vector to be the self arg. Maybe it should be a global function or thrown into util.

Reply:
Should I remove it then?

Reply:
I mean, it's your PR, it's ultimately up to Rubat. I just think it's vague implemented as a vector class method personally.

Reply:
You are probably right ;)